src/main/java/uk/gov/register/store/DataAccessLayer.java

```diff
@@ -22,35 +19,38 @@
     Optional<Entry> getEntry(int entryNumber);
     Collection<Entry> getEntries(int start, int limit);
     Collection<Entry> getAllEntries();
+
     Iterator<Entry> getEntryIterator();
     Iterator<Entry> getEntryIterator(int totalEntries1, int totalEntries2);
     Iterator<Entry> getIndexEntryIterator(String indexName);
     Iterator<Entry> getIndexEntryIterator(String indexName, int totalEntries1, int totalEntries2);
     <R> R withEntryIterator(Function<EntryIterator, R> callback);
     int getTotalEntries();
     int getTotalEntries(EntryType entryType);
+    int getTotalStagedEntries(EntryType entryType);
```

Comment:
Can we remove the need for this? It would be good to keep the knowledge of staged data below this interface.